hoomd/mpcd/CollisionMethod.cc

```diff
@@ -33,6 +38,38 @@ mpcd::CollisionMethod::CollisionMethod(std::shared_ptr<SystemDefinition> sysdef,
         uint64_t multiple = cur_timestep / m_period + (cur_timestep % m_period != 0);
         m_next_timestep = multiple * m_period + phase;
         }
+
+#ifdef ENABLE_HIP
+    if (m_exec_conf->isCUDAEnabled())
+        {
+        m_store_tuner.reset(new Autotuner<1>({AutotunerBase::makeBlockSizeRange(m_exec_conf)},
+                                             m_exec_conf,
+                                             "mpcd_rigid_store"));
+        m_drawrandvec_tuner.reset(new Autotuner<1>({AutotunerBase::makeBlockSizeRange(m_exec_conf)},
+                                                   m_exec_conf,
+                                                   "mpcd_rigid_rand"));
+        m_netvelo_tuner.reset(new Autotuner<1>({AutotunerBase::makeBlockSizeRange(m_exec_conf)},
+                                               m_exec_conf,
+                                               "mpcd_rigid_netvelo"));
+        m_applyrandvec_tuner.reset(
+            new Autotuner<1>({AutotunerBase::makeBlockSizeRange(m_exec_conf)},
+                             m_exec_conf,
+                             "mpcd_rigid_applyrand"));
+        m_accumulate_tuner.reset(new Autotuner<1>({AutotunerBase::makeBlockSizeRange(m_exec_conf)},
+                                                  m_exec_conf,
+                                                  "mpcd_rigid_accum"));
+        m_transfer_tuner.reset(new Autotuner<1>({AutotunerBase::makeBlockSizeRange(m_exec_conf)},
+                                                m_exec_conf,
+                                                "mpcd_rigid_transfer"));
+        m_autotuners.insert(m_autotuners.end(),
+                            {m_store_tuner,
+                             m_drawrandvec_tuner,
+                             m_netvelo_tuner,
+                             m_applyrandvec_tuner,
+                             m_accumulate_tuner,
+                             m_transfer_tuner});
+        }
+#endif // ENABLE_HIP
     }
 
 void mpcd::CollisionMethod::collide(uint64_t timestep)
```

Comment:
An error needs to be thrown somewhere in this method if a temperature is required but not set. It would need to come after you check the collision warnings, since that may modify whether a temperature is needed.

Reply:
Could it be thrown by `requireTemperature`? Then collision warnings could call it if there's a rigid body particle in the embedded particles.

Reply:
No, `requireTemperature` is a method to require the temperature object at runtime, but you need to defer the check until the collision happens. The collision warning method uses that method to require the temperature. Just add something like:

```
if (m_needs_temperature && !m_T)
    {
    throw std::runtime_error("Temperature required by collision method");
    }
```

right after you check the warnings.